packages/services/src/balances/bitcoin-balances.service.ts

```diff
@@ -0,0 +1,101 @@
+import { BtcCryptoAssetBalance } from '@leather.io/models';
+import {
+  baseCurrencyAmountInQuote,
+  createBtcCryptoAssetBalance,
+  createMoney,
+  isDefined,
+  sumNumbers,
+} from '@leather.io/utils';
+
+import { LeatherApiClient, LeatherApiUtxo } from '../infrastructure/api/leather/leather-api.client';
+import { MarketDataService } from '../market-data/market-data.service';
+import { hasUneconomicalBalance, removeExistingUtxos } from './bitcoin-balances.utils';
+
+export interface BtcBalance {
+  balanceBtc: BtcCryptoAssetBalance;
+  balanceUsd: BtcCryptoAssetBalance;
+}
+
+export interface BitcoinBalancesService {
+  getBtcBalance(descriptors: string[], signal?: AbortSignal): Promise<BtcBalance>;
+}
+
+export function createBitcoinBalancesService(
+  leatherApiClient: LeatherApiClient,
+  marketDataService: MarketDataService
+): BitcoinBalancesService {
+  /**
+   * Retrieves total BTC balance for listed descriptors. Includes all sub-balances (inbound, outbound, protected, uneconomical).
+   *
+   * @returns {BtcBalance} BTC balance denominanted in both in BTC and fiat (USD).
+   */
+  async function getBtcBalance(descriptors: string[], signal?: AbortSignal) {
+    const utxoPromises = descriptors.map(descriptor =>
+      leatherApiClient.fetchUtxos(descriptor, signal)
+    );
+    const totalUtxos = (await Promise.all(utxoPromises)).flat().filter(isDefined);
+    const totalBalanceBtc = createMoney(
+      sumNumbers(totalUtxos.map(utxo => Number(utxo.value))).toNumber(),
+      'BTC'
+    );
+    // TODO: fetch inbound utxos
+    const inboundUtxos = await getInboundUtxos(descriptors);
+    const inboundBalanceBtc = createMoney(
+      sumNumbers(inboundUtxos.map(utxo => Number(utxo.value))).toNumber(),
+      'BTC'
+    );
+    // TODO: fetch outbound utxos
+    const outboundUtxos = await getOutboundUtxos(descriptors);
+    const outboundBalanceBtc = createMoney(
+      sumNumbers(outboundUtxos.map(utxo => Number(utxo.value))).toNumber(),
+      'BTC'
+    );
+    // TODO: fetch protected utxos
+    const protectedUtxos = await getProtectedUtxos(descriptors);
+    const protectedBalanceBtc = createMoney(
+      sumNumbers(protectedUtxos.map(utxo => Number(utxo.value))).toNumber(),
+      'BTC'
+    );
+    const uneconomicalUtxos = removeExistingUtxos(totalUtxos, protectedUtxos).filter(
+      hasUneconomicalBalance
+    );
+    const uneconomicalBalanceBtc = createMoney(
+      sumNumbers(uneconomicalUtxos.map(utxo => Number(utxo.value))).toNumber(),
+      'BTC'
+    );
+
+    const btcMarketData = await marketDataService.getBtcMarketData(signal);
+    return {
+      balanceBtc: createBtcCryptoAssetBalance(
+        totalBalanceBtc,
+        inboundBalanceBtc,
+        outboundBalanceBtc,
+        protectedBalanceBtc,
+        uneconomicalBalanceBtc
+      ),
+      balanceUsd: createBtcCryptoAssetBalance(
```

Comment:
In the future we want to support multi currencies but that setting is not connected yet and [I started deliberately labelling balances](https://github.com/leather-io/mono/pull/590/files#r1825916034) with `USD` to make this clearer as in `i18nFormatCurrency` we have it hardcoded to only accept USD anyway:
```
export function i18nFormatCurrency(quantity: Money, decimals: number = 2) {
  if (quantity.symbol !== 'USD') throw new Error('Cannot format non-USD amounts');
  const currencyFormatter = new Intl.NumberFormat('en-US', {
```

I agree it should be more scalable and offer alternative currencies but that's a whole other initiative and for now I think `balanceUsd` is more apt as it can't be anything else

Reply:
@alexp3y - is the market data currently only supporting USD anyway? 

Just thinking we could add a project for updating both `extension` + `mobile` to support multi currency in the future. (Probably far in the future)